### DISCUSSION TITLE
Palette attribute Gittree

### DISCUSSION THREAD

**Discussion by m-vik** [Kitware/trame#246] [Q&A] [answered]
Hello,
I am trying to customize the colors of the gittree component with the palette attribute. 
```python
trame.GitTree(
        sources=(
            "pipeline",
            tree,
        ),
        actives_change=(actives_change, "[$event]"),
        visibility_change=(visibility_change, "[$event]"),
        font_size=10,
        delta_x=10,
        # delta_y=30,
        # radius=5,
        # palette=['#e1002a', '#417dc0', '#1d9a57', '#e9bc2f', '#9b3880', '#9b3880']
        palette=["#F44336", "#F44336", "#F44336", "#F44336", "#F44336"]
    )
```
However when I try to set this value, the browser open, but rending no ui at all. When I comment out the attribute everything works well.
Is the input wrong?

**Comment by jourdain** [comment, marked as answer]
You have those 2 options
- Pure JS: `palette=("['#F44336', '#F44336', '#F44336', '#F44336', '#F44336']",),`
- Use a variable: `palette=("palette", ["#F44336", "#F44336", "#F44336", "#F44336", "#F44336"]),`
- Not sure that works: `palette=([...],),`

**Reply by m-vik** [reply]
I tried this, it is not working.
```python
palette=('["#F44336", "#F44336", "#F44336", "#F44336", "#F44336"]',),
```
I am getting this in the drawer:
![error](https://user-images.githubusercontent.com/121301781/234319451-96944a43-a175-4c7c-b559-5f1e66c9a4ac.PNG)

**Reply by jourdain** [reply]
Updated my comment. Basically the `"` are conflicting with the generated HTML `attr="["a", "b"]"`

**Reply by m-vik** [reply]
> You have those 2 options
> 
> * Pure JS: `palette=("['#F44336', '#F44336', '#F44336', '#F44336', '#F44336']",),`
> * Use a variable: `palette=("palette", ["#F44336", "#F44336", "#F44336", "#F44336", "#F44336"]),`
> * Not sure that works: `palette=([...],),`

The second and the third one is working. Thank you! :)

**Comment by jourdain** [comment]
Look at the course reference slides

**Reply by m-vik** [reply]
Thank you I will do.
